go/worker/common/committee/group.go

```diff
@@ -440,18 +444,21 @@ func (g *Group) HandlePeerMessage(unusedPeerID signature.PublicKey, msg *p2p.Mes
 		switch {
 		case msg.GroupVersion < g.activeEpoch.groupVersion:
 			// Stale messages will never become valid.
-			return nil, p2pError.Permanent(fmt.Errorf("group version in the past"))
+			return p2pError.Permanent(fmt.Errorf("group version in the past"))
 		case msg.GroupVersion > g.activeEpoch.groupVersion:
 			// Messages from the future may eventually become valid.
-			return nil, fmt.Errorf("group version from the future")
+			return fmt.Errorf("group version from the future")
 		}
 
-		return g.activeEpoch.roundCtx, nil
+		return nil
 	}()
 	if err != nil {
 		return err
 	}
 
+	ctx, cancel := context.WithTimeout(context.Background(), peerMessageProcessTimeout)
+	defer cancel()
```

Comment:
what's the context of this commit? why do we now want to allow handling of messages while we're not in a round?

also, why `defer cancel()` here? is it know that `HandlePeerMessage` doesn't do anything after returning?

Reply:
Yeah the previous context that was used was actually incorrect. `HandlePeerMessage` should dispatch longer-lasting operations to background goroutines with their own contexts (all the current implementations do). I'll add an explicit comment to the interface.

Reply:
Done.

Reply:
thanks for adding that info